slack_global_report.py

```diff
@@ -0,0 +1,241 @@
+        stats = ur['statistics']
+        posters = int(stats['posters'])
+        active_users = stats['active_users']
+        percent = (posters * 100.0) / active_users
+        text = "*{:,}/{:,}* (or *{:.1f}%*) users posted messages\n".format(posters, active_users, percent)
+        text += "Median message count was *{}*\n".format(stats['median messages'])
+        text += "The top ten posters contributed *{:.1f}%* of all messages (lower is better)\n".format(stats['topten messages'])
+        text += "The top *{}* posters (higher is better) accounted for about 50% of total volume\n".format(stats['50percent of words'])
+        blocks.append(self.sf.text_block(text))
+        w = stats['words']
+        m = stats['messages']
+        # it = interim text
+        it = "People posted *{:,}* words in *{:,}* messages (approximately *{:,}* pages), "
+        it += "or about *{:.1f}* words per message, *{:.1f}* words per poster, "
+        it += "or *{:.1f}* messages per poster"
+        text = it.format(int(w), int(m), int(w / 500), w/m, w/posters, m/posters)
+        blocks.append(self.sf.text_block(text))
+        blocks.append(self.sf.divider())
+        return blocks
+
+    def top_channels(self, ur, pur):
+        blocks = []
+        top = 20
+        header = "*Top {} Channels*".format(top)
+        blocks.append(self.sf.text_block(header))
+        channels = ur['channels']
+        cids = list(channels.keys())[:top]
+        cinfo = ur['channel_info']
+        cstats = ur['channel_stats']
+        cusers = ur['channel_user']
+        for idx, channel in enumerate(cids):
+            it = "{}. {} ".format(idx + 1, self.sf.show_cid(channel))
+            ci = cinfo[channel]
+            cs = cstats[channel]
+            cu = cusers[channel]
+            if ci['new']:
+                it += " (new)"
+            it += "{:,} members ".format(ci['members'])
+
+            m = channels[channel][0]
+            w = channels[channel][1]
+            p = len(cu)
+            it += "*{:,}* posters *{:,}* words *{:,}* messages ".format(p, w, m)
+            it += "*{:.1f}* words/poster ".format(w/p)
+            it += "*{:.1f}%* of total traffic, ".format(cs['percent'])
+            it += "*{:.1f}%* cumulative of total ".format(cs['cpercent'])
+            blocks.append(self.sf.text_block(it))
+        blocks.append(self.sf.divider())
+        return blocks
+
+    def top_users(self, ur, pur):
+        blocks = []
+        top = 20
+        header = "*Top {} Users*".format(top)
+        blocks.append(self.sf.text_block(header))
+        stats = ur['statistics']
+        us = ur['user_stats']
+        uids = stats['50percent users for words'][:top]
+        for idx, uid in enumerate(uids):
+            usu = us[uid]
+            m = usu['count'][0]
+            w = usu['count'][1]
+            per = usu['percent_of_words']
+            cper = usu['cum_percent_of_words']
+            rphw = usu['reactions'] * 100.0 / w
```

Comment:
Can you add some copy at the top of this section that explains what rphw is?

Reply:
Added